### DISCUSSION TITLE
pg_dump/pg_restore crashes with parallel jobs after macOS Sequoia update

### DISCUSSION THREAD

**Discussion by matjazs** [Homebrew/discussions#5759] [Everyday usage] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.4.6-35-gc329e72
ORIGIN: https://github.com/Homebrew/brew
HEAD: c329e726f8f301c700629ec538c659246d841d7c
Last commit: 19 hours ago
Core tap JSON: 22 Nov 08:28 UTC
Core cask tap JSON: 22 Nov 08:28 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: vim
HOMEBREW_MAKE_JOBS: 12
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.3.6 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.6/bin/ruby
CPU: dodeca-core 64-bit arm_blizzard_avalanche
Clang: 16.0.0 build 1600
Git: 2.39.5 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 15.1.1-arm64
CLT: 16.1.0.0.1.1729049160
Xcode: 16.1
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

Is anyone else experiencing crashes of `pg_dump` or `pg_restore` when using parallel jobs (`--jobs=n` where n > 1) with the latest `postgresql@14` installed via Homebrew, following the recent Sequoia update?

The commands work fine with `--jobs=1`, but using higher values leads to crashes. My system logs and crash reports mention denied access to the system keychain: `client doesn't have entitlement for system keychain`. I’m trying to determine if this is a widespread issue or something specific to my setup.

Would appreciate any insights or suggestions!

**Comment by hughsimpson** [comment]
This has been happening to me for at least a week or so. Are you restoring to a pg instance running inside docker? Was wondering if that might be related.

**Reply by matjazs** [reply]
No, I was restoring to an instance running directly on the host machine and to a GCP Cloud SQL instance, same error.

**Reply by hughsimpson** [reply, marked as answer]
Seems to be an issue with GSS. My local restores have stopped crashing when I set `POSTGRES_GSSENCMODE=DISABLE`

**Reply by matjazs** [reply]
Thanks for the suggestion! Your workaround worked perfectly; I just had to use `PGGSSENCMODE=disable`.
